corehq/apps/case_search/xpath_functions/comparison.py

```diff
@@ -18,18 +23,35 @@ def property_comparison_query(context, case_property_name_raw, op, value_raw, no
 
     case_property_name = serialize(case_property_name_raw)
     value = unwrap_value(value_raw, context)
-
+    # adjust the user's input date based on project timezones
+    is_user_input = False
+    try:
+        # this might be inconsistent in daylight savings situations
+        value = adjust_input_date_by_timezone(value_to_date(node, value),
+                                              get_timezone_for_domain(context.domain), op)
+        is_user_input = True
+    except (XPathFunctionException, AssertionError):
+        # AssertionError is caused by tests that use domains without a valid timezone (in get_timezeone_for_domain)
+        pass
     if op in [EQ, NEQ]:
         query = case_property_query(case_property_name, value, fuzzy=context.fuzzy)
         if op == NEQ:
             query = filters.NOT(query)
         return query
     else:
         try:
-            return case_property_range_query(case_property_name, **{RANGE_OP_MAPPING[op]: value})
+            return case_property_range_query(case_property_name, is_user_input=is_user_input,
+                                             **{RANGE_OP_MAPPING[op]: value})
         except (TypeError, ValueError):
             raise CaseFilterError(
                 gettext("The right hand side of a comparison must be a number or date. "
                   "Dates must be surrounded in quotation marks"),
                 serialize(node),
             )
+
+
+def adjust_input_date_by_timezone(date, timezone, op):
```

Comment:
This change seems to have caused this [bug](https://dimagi-dev.atlassian.net/jira/servicedesk/projects/SUPPORT/queues/custom/81/SUPPORT-17192) for date case properties. I.E if the xpath query is`dob="1984-08-15"`, "1984-08-15" is being converted to some other value even though dob is stored in ES as "1984-08-15". The filter is built with the converted value, leading to no matches.

I've been thinking about what the fix would be and i'm a bit stumped. The only thing i've thought of: to know whether to apply to conversion, we need to know if the property being filtered is stored as UTC. A way to do that would be to only convert `CommCareCase` fields of type `DateTimeField`. But that doesn't sounds like a great strategy. 
Commenting to see if anything else might come to mind that I'm not thinking of.

Reply:
Sorry I didn't get a chance to look at this properly on Friday but yea this is absolutely my bad. The error I was tackling described date ranges so testing the equal operator somehow never crossed my mind. That being said, I think a simple fix off the top of my head might be just to put a conditional that says ignore the timezone conversion logic if '=' is present. Not the most elegant thing but I'd be happy to work with you more on this.

Reply:
Oh nevermind I just saw the other PR

Reply:
Thanks Minha! I should have updated this comment earlier - https://github.com/dimagi/commcare-hq/pull/33272